### PR TITLE
register ATen fallback for instance_norm2d gradient

### PR DESCRIPTION
**Pull request by prathikr**
### Description
We currently have an inference kernel for Instance Normalization but no gradient kernel. Falling back to ATen for now to verify stable diffusion model works with this gradient before investing time in writing a cuda kernel for this op.

### Motivation and Context
Instance Normalization gradient required for [huggingface's diffusers library](https://github.com/huggingface/diffusers)

### CODE REVIEW

**Review by baijumeswani — COMMENTED** [review]
looks good. Can we add an ORTModule test?

File: orttraining/orttraining/python/training/ortmodule/_custom_gradient_registry.py [hunk 1]
```diff
@@ -125,7 +137,6 @@ def max_pool2d_gradient():
         ),
     ]
 
-
```
Comment (hanbitmyths) [inline orttraining/orttraining/python/training/ortmodule/_custom_gradient_registry.py:128]: Does this cause a Lint error?

**Comment by iK1D** [comment]
ATen fallback is not working like this. In your code, you register gradient of ATen(operator="instance_norm2d") to ATen(operator="instance_norm2d_backward"). But your forward graph actually doesn't have any ATen(operator="instance_norm2d") node. From the description, it's exported to an ONNX's InstanceNormalization node.

All available aten functions from PyTorch we can call is from https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml, neither instance_norm2d nor instance_norm2d_backward is in the list. It means both ATen(operator="instance_norm2d") and ATen(operator="instance_norm2d_backward") are invalid nodes.

I am guessing the forward function is aten::instance_norm. PyTorch even doesn't have a corresponding backward function for it. Inside the implementation, it calls aten::batch_norm, and PyTorch's autograd will use batch_norm's gradient to compose the gradient sub-graph. This means it's not an easy job to use ATen fallback for the gradient of aten::instance_norm/InstanceNormalization (it's doable, but not just one ATen node).

Since InstanceNormalization is actually BatchNorm with some other extra handlings, we may need to look into the implementation and think of how we can use BatchNormGrad and some other Ops to compose the gradient.

**Comment by iK1D** [comment]
I checked the diffusers model code and found that it's torch.nn.GroupNorm that introduces the ONNX's InstanceNormalization. I created another PR https://github.com/microsoft/onnxruntime/pull/13293 to use ATen fallback to support this. Please have a try.

**Comment by baijumeswani** [comment]
Is this pull request still relevant?
